text/0000-pack-publish-buildpack.md

```diff
@@ -0,0 +1,287 @@
+
+The user will run the `pack yank-buildpack <namespace>/<name>@<version>` command to yank a buildpack. This command will create a Github Issue on the registry. If a buildpack author wants to undo the yank and make the buildpack version available in the index, they can run the same command and use the `--undo` flag.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The implementation of adding and yanking buildpacks will require new configuration elements in Pack's config file, and two new commands in Pack.
+
+## Configuration
+
+A default registry will be configured in `.pack/config.toml` with the following schema.
+
+##### Base Schema
+```toml
+default-registry="<string>"
+
+[[registries]]
+name="<string>"
+type="<string>"
+# <additional per "type" config entries>
+```
+
+The `default-registry` value must match one of the tables in the `[[registries]]` array with the same `name`. The `name` value can be any string.
+
+The `type` key in the `[[registries]]` table can be set to one of two values, described below:
+
+##### Type: GitHub Schema
+```toml
+[[registries]]
+name="<string>"
+type="github"
+url="<url to git repo>"
+issues-url="<url to issues for registering>" # optional, default: "<url>/issues"
+```
+
+##### Type: Git Schema
+```toml
+[[registries]]
+name="<string>"
+type="git"
+url="<path to git repo>"
+```
+
+#### Default
+
+The following will be the implied default when no registries are present, however, it can be overridden by defining a registry with the name "official".
+
+```toml
+default-registry="official"
+
+[[registries]]
+name="official"
+type="github"
+url="https://github.com/buildpacks/registry"
+issues-url="https://github.com/buildpacks/registry/issues"
+```
+
+## Format
+
+### Type: Git
+
+A commit would be created with the following message:
+
+```
+[<ADD|YANK>] <namespace>/<name>@<version>
+
+<user message (optional)>
+```
+
+##### Example
+
+```
+[YANK] buildpacks-io-samples/hello-universe@0.0.1
+
+Major bug issues found. See https://github.com/buildpacks/samples/issues.
+```
+
+### Type: GitHub issue
+
+An issue would be created with the following content:
+
+- Title: `[<ADD|YANK>] <namespace>/<name>@<version>`
+- Body:
+
+        <user message (optional)>
+
+        ### Data
+
+        ```toml
+        id = <ns/name (required)>
+        version = <version (required)>
+        addr = <address-to-image (only required for ADD)>
+        digest = <digest-of-image (only required for ADD)>
+        ```
+
+##### Example
+
+- Title: `[YANK] samples/hello-universe@0.0.1`
+- Body:
+
+        Major bug issues found. See https://github.com/buildpacks/samples/issues.
+
+        ### Data
+
+        ```toml
+        id = samples/hello-universe
+        version = 0.0.1
+        ```
+
+## `pack register-buildpack <url>`
+
+Arguments:
+
+* `url-or-tag` - the location of a buildpackage _in a Docker registry_ to be released in the buildpack registry
```

Comment:
In other documents we refer to this as the the `image-name`. Is the something special, or additional, when we say "url OR tag"?

Reply:
addressed in cc49f2f